src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -202,6 +202,89 @@ Single branch
                         continue;
                     }
 
+                    /*
+                        Handle try/catch blocks inside async state machine
+                     */
+                    if (isAsyncStateMachineMoveNext)
+                    {
+                        /*
+                            Typical generated code for catch block (inside async state machine MoveNext() method) 
+                            is this:
+
+                            catch ...
+		                    {
+			                    // (no C# code)
+			                    IL_0028: stloc.2
+			                    // object obj2 = <>s__1 = obj;
+			                    IL_0029: ldarg.0
+			                    // (no C# code)
+			                    IL_002a: ldloc.2
+			                    IL_002b: stfld object ...::'<>s__1'
+			                    // <>s__2 = 1;
+			                    IL_0030: ldarg.0
+			                    IL_0031: ldc.i4.1
+			                    IL_0032: stfld int32 ...::'<>s__2'      <- store 1 into <>s__2
+			                    // (no C# code)
+			                    IL_0037: leave.s IL_0039
+		                    } // end handle
+
+                            // int num2 = <>s__2;
+                            IL_0039: ldarg.0
+		                    IL_003a: ldfld int32 ...::'<>s__2'          <- load <>s__2 value and check if 1
+		                    IL_003f: stloc.3
+                            // if (num2 == 1)
+		                    IL_0040: ldloc.3
+		                    IL_0041: ldc.i4.1
+		                    IL_0042: beq.s IL_0049                      <- if <>s__2 value is 1 go to exception handler code
+
+                            IL_0044: br IL_00d6
+
+                            IL_0049: nop                                <- start exception handler code
+                            ...
+
+                            
+                            So starting from branch instruction 'beq.s', we can go back to starting block instruction 
+                            which is always 5 step before and then check if this istruction is the end of an exception handler block
+                        */
+
+                        int branchIndex = instructions.IndexOf(instruction);
```

Comment:
Can you use binary search?IL instruction offset are sequential ordered O(n) -> O(log n) similar https://github.com/tonerdo/coverlet/blob/master/src/coverlet.core/Symbols/CecilSymbolHelper.cs#L448